home6.1/MessageDispatcher.java

```diff
@@ -0,0 +1,45 @@
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+
+public class MessageDispatcher {
+    private Map<UUID, MessageSubscriber> subscribers;
+
+    public MessageDispatcher() {
+        this.subscribers = new ConcurrentHashMap<>();
+    }
+
+
+    public UUID subscribe(MessageSubscriber subscriber) {
```

Comment:
we should subscribe with some function object therefore subscriber will be able to provide for example lambda that will do the handling or few callbacks can be registered by a single client class.